core/src/main/java/apoc/load/Xml.java

```diff
@@ -151,16 +185,23 @@
         }
     }
 
-    private XMLStreamReader getXMLStreamReaderFromUrl(String url, XmlImportConfig config) throws IOException, XMLStreamException {
-        apocConfig.checkReadAllowed(url);
-        url = FileUtils.changeFileUrlIfImportDirectoryConstrained(url);
-        URLConnection urlConnection = new URL(url).openConnection();
-        FACTORY.setProperty(XMLInputFactory.IS_COALESCING, true);
-        InputStream inputStream = urlConnection.getInputStream();
-
+    private XMLStreamReader getXMLStreamReader(Object urlOrBinary, XmlImportConfig config) throws IOException, XMLStreamException {
+        InputStream inputStream;
+        if (urlOrBinary instanceof String) {
+            String url = (String) urlOrBinary; 
+            apocConfig.checkReadAllowed(url);
+            url = FileUtils.changeFileUrlIfImportDirectoryConstrained(url);
+            URLConnection urlConnection = new URL(url).openConnection();
+            inputStream = urlConnection.getInputStream();
+        } else if (urlOrBinary instanceof byte[]) {
+            inputStream = getInputStreamFromBinary((byte[]) urlOrBinary, config.getCompressionAlgo());
+        } else {
+            throw new RuntimeException(ERROR_BYTES_OR_STRING);
+        }
         if (config.isFilterLeadingWhitespace()) {
             inputStream = new SkipWhitespaceInputStream(inputStream);
         }
+        FACTORY.setProperty(XMLInputFactory.IS_COALESCING, true);
```

Comment:
non ho capito questo a cosa serve? se lo cambi qui in un qualunque momento lo stai cambiando globalmente... a quest punto conviene metterlo dentro un blocco `static` no=